arm9/overlays/24/src/ov24_02254840.c

```diff
@@ -0,0 +1,254 @@
+#include "global.h"
+#include "bg_window.h"
+#include "unk_0200CA44.h"
+#include "gf_gfx_loader.h"
+#include "overlay_24.h"
+
+extern void Poketch_InitApp(void *func1, void *func2);
+extern u32 ov20_02254130(void *, u32, void *, void*, u32);
+extern void ov20_02254198(u32);
+extern void ov20_02252C14(u32, u32);
+extern void ov20_022529A0(u32);
+extern void ov20_0225298C(u32);
+extern BOOL ov20_02252C08(u32);
+extern BOOL TouchScreen_GetTapState(u32*, u32*);
+
+static void ov24_02254840(void)
+{
+    Poketch_InitApp(ov24_02254854, ov24_02254990);
+}
+
+#define NitroStaticInit ov24_02254840
+#include "sinit.h"
+
+BOOL ov24_02254854(MemoPadAppHandler** appHandlerOut, int arg1, int arg2, int arg3) {
+    MemoPadAppHandler* appHandler = AllocFromHeap(HEAP_ID_POKETCH_APP, sizeof(MemoPadAppHandler));
+    if (appHandler) {
+        if (ov24_0225489C(appHandler, arg1, arg2, arg3) != 0) {
+            if (SysTask_CreateOnMainQueue((SysTaskFunc)ov24_02254918, appHandler, 1) != 0) {
+                *appHandlerOut = appHandler;
+                return TRUE;
+            }
+        }
+        FreeToHeap(appHandler);
+    }
+    return FALSE;
+}
+
+BOOL ov24_0225489C(MemoPadAppHandler* appHandler, u32 arg1, u32 arg2, u32 arg3) {
+    static const u8 ov24_022550F8[] = {
+        0x18, 0x58, 0xB4, 0xCC, 0x68, 0xA8, 0xB4, 0xCC
+    };
+    appHandler->drawState.unk16E8 = arg3;
+    appHandler->drawState.stylusType = STYLUS_TYPE_DRAW;
+    if (ov24_02254CA0(&(appHandler->displayHandler), &(appHandler->drawState))) {
+        appHandler->unk0 = 0;
+        appHandler->unk1 = 0;
+        appHandler->unk2 = 0;
+        appHandler->unk3 = 0;
+        appHandler->unk16FC = ov20_02254130(ov24_022550F8, 2, ov24_02254960, appHandler, 8);
+        appHandler->unk16F8 = arg1;
+        return TRUE;
+    }
+    return FALSE;
+}
+
+void ov24_022548F4(MemoPadAppHandler* appHandler) {
+    ov20_02254198(appHandler->unk16FC);
+    ov24_02254D48(appHandler->displayHandler);
+    FreeToHeap(appHandler);
+}
+
+void ov24_02254918(void* arg0, MemoPadAppHandler* appHandler) {
+    static BOOL (*const ov24_02255100[3])(MemoPadAppHandler*) = {
```

Comment:
the brackets here seem kinda weird for me, I'm not sure why that is but they feel strange

Reply:
I'm not sure exactly what is meant by this but I generally find function array declarations strange